overlord/snapstate/handlers.go

```diff
@@ -605,6 +605,7 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	if snapsup.Required { // set only on install and left alone on refresh
 		snapst.Required = true
 	}
+	snapst.UserID = snapsup.UserID
```

Comment:
we should set it only if it's different from 0 I think (otherwise with the rest of the code a refresh all will forget the value)